kotlin/src/main/com/looker/rtl/Transport.kt

```diff
@@ -267,100 +242,174 @@ class Transport(val options: TransportOptions) {
         } + addQueryParams(path, queryParams)
     }
 
-    inline fun <reified T> request(
+    private fun getAllTrustingVerifiers(): Pair<SSLSocketFactory, HostnameVerifier> {
+        // NOTE! This is completely insecure and should ONLY be used with local server instance
+        // testing for development purposes
+        val tm: X509TrustManager = object : X509TrustManager {
+            override fun getAcceptedIssuers(): Array<X509Certificate?> {
+                return arrayOfNulls(0)
+            }
+
+            @Throws(CertificateException::class)
+            override fun checkClientTrusted(
+                certs: Array<X509Certificate?>?,
+                authType: String?,
+            ) {
+            }
+
+            @Throws(CertificateException::class)
+            override fun checkServerTrusted(
+                certs: Array<X509Certificate?>?,
+                authType: String?,
+            ) {
+            }
+        }
+        val trustAllCerts = arrayOf(tm)
+        val sslContext = SSLContext.getInstance("SSL")
+        sslContext.init(null, trustAllCerts, SecureRandom())
+
+        val sslSocketFactory: SSLSocketFactory = sslContext.socketFactory
+        val hostnameVerifier = HostnameVerifier { _, _ -> true }
+
+        return Pair(sslSocketFactory, hostnameVerifier)
+    }
+
+    /** Given [TransportOptions], selects the requested [HttpTransport].
+     *
+     * Will disable SSL certificate verification iff [TransportOptions.verifySSL] is false.
+     */
+    fun initTransport(options: TransportOptions): HttpTransport {
+        return when (HttpTransports.valueOf(options.httpTransport.uppercase())) {
+            HttpTransports.APACHE -> {
+                // TODO: fix bug upstream that does not pass client context to requests.
+                //  The Netscape `expire` datetime format is not compatible with the "default" and
+                //  will log invalid warnings.
+                val clientBuilder =
```

Comment:
The response cookies we get back from Looker use the ` Wdy, DD-Mon-YY HH:MM:SS GMT` format for `expires`. We should be able to push down the config here to the low-level Apache request but the request does not reference the client context. I believe it is a bug in the Google Java Client.

Note, this doesn't stop requests, but rather logs a confusing error. We still get a correct response back. 

Alternatively, could we use `disableCookieManagement()` so the low-level client doesn't try to validate cookies. The SDK usually shouldn't care about cookies right? Wouldn't the required auth token header make cookies redundant?

Reply:
We definitely have responses that would set cookies and those would be needed (the ironically named cookieless embed is one example) but this could be a case of worrying about it when we discover someone needs it. The only concern I have is whether the Looker Android mobile client reads any cookies from Looker, but it probably doesn't? I've not really looked at how cookies are used in non-browser situations and someone could implement a custom payload parser if it really was required as long as we have the hooks available to get the original request. 

Do we have a test that verifies a payload with missing "required" fields still gets parsed correctly? Or are you still using GSon for parsing with whichever client? I'll look at the code now that I'm on a bigger screen.